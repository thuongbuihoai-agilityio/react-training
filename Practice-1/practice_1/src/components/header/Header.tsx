import React from "react";
import Navigation from "../navigation/Menu";
import { MenuType } from "../../types/menu";
import SectionTitle from "../title/SectionTitle";
import "./header.css";
import { Description } from "../description/Description";
import Button from "../button/Button";

interface HeaderProps {
  MAIN_TEXT: string;
  menuList: MenuType[]
  menuBar: string;
}

export default function Header({
  MAIN_TEXT,
  menuList,
  menuBar
}: HeaderProps): JSX.Element {
  return (
    <header className="header">
      <div className="header__menu">
        <SectionTitle className="nav__title">
          <span>HOFMANN</span>
        </SectionTitle>
        <Navigation menuList={menuList} menuBar={menuBar}></Navigation>
      </div>
      <div className="container">
        <div className="header__info">
          <div className="header__content">
            <SectionTitle className="main__title">
              <span>What Makes A Hotel Boutique</span>
            </SectionTitle>
            <Description className="main">{MAIN_TEXT}</Description>
            <Button className="primary">Explore</Button>
          </div>
        </div>
      </div>
    </header>
  );
}
