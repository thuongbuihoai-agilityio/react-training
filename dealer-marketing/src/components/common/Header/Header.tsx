import { MENU_CONTACT } from "@src/constants/menu";
import React from "react";
import Menu from "../Menu/Menu";
import headerStyle from "./header.module.css";

const Header = () => {
  return (
    <header className={headerStyle.header}>
      <Menu type="light" menuList={MENU_CONTACT} />
    </header>
  );
};

export default Header;
