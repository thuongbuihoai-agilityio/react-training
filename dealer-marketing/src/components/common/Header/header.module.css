.header {
  padding: 11px 100px;
  background-color: black;
}

.header-contact {
  display: flex;
  flex-direction: row-reverse;
}
