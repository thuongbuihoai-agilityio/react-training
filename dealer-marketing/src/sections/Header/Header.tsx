import React from "react";
import Menu from "../../components/common/Menu/Menu";
import { MENU_CONTACT } from "@constants/menu";
import styleHeader from "./header.module.css";

const Header = () => {
  return (
    <header className={styleHeader.header}>
      <div className={styleHeader["header-contact"]}>
        <Menu type="light" menuList={MENU_CONTACT} />
      </div>
    </header>
  );
};

export default Header;
