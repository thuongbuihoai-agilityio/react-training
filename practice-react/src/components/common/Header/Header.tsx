import React from "react";
import Logo from "../Logo/Logo";
import url from "../../../assets/images/logos/logo.png";
import Menu from "../Menu/Menu";
import MENU_LIST from "@constants/menu";
import Button from "../Button/Button";
import Text from "../Text/Text";

const Header: React.FC = () => {
  return (
    <header data-testid="header" className="header">
      <Logo src={url} />
      <div className="header__about">
        <Menu menuList={MENU_LIST} />
        <Button text="Sign Up" type="light" />
        <Button text="Login" type="outline--light" />
      </div>
      <Text text="Lorem ipsum dolor sit" type="blur" />
      <h1 className="header__title">Lorem ipsum dolor sit amet, consectetur adipiscing elit.</h1>
      <Button text="Sign Up" type="large" />
    </header>
  );
};

export default Header;
