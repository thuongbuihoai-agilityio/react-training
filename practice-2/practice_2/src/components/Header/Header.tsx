import React from "react";
import { users } from "../../constants/header";
import { HeaderProps } from "../../types/header";
import Logo from "../Logo/Logo";
import Search from "../Search/Search";
import User from "../Icon/Icon";
import "./header.css"

export default function Header({url}: HeaderProps) {
  return (
    <header className="header">
      <div className="header__logo">
        <Logo href="#Home" src={url} />
      </div>
      <Search />
      <div className="header__user">
        <User
          username={users.username}
          avatar={users.avatar}
        />
      </div>
    </header>
  );
}
