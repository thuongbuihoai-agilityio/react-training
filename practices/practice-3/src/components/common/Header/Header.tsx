import React, { memo } from "react";
import Logo from "../Logo/Logo";
import url from "@/assets/images/logos/logos.png";
import backgroundUrl from "@/assets/images/backgrounds/slide-1.jpg";
import Menu from "../Menu/Menu";
import MENU_LIST from "@/constants/menu";
import Button from "../Button/Button";
import "./header.css";

const Header: React.FC = memo(() => {
  return (
    <>
      <header data-testid="header" className="header">
        <div className="header__logo">
          <Logo src={url} />
        </div>
        <div className="header__about">
          <Menu menuList={MENU_LIST} />
          <Button className="btn btn__primary" text="ORDER ONLINE" />
        </div>
      </header>
      <figure>
        <img src={backgroundUrl} className="header__img" alt="" />
      </figure>
    </>
  );
});

export default Header;
