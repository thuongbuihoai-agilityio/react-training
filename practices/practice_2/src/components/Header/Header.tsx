import React from "react";
import { HeaderProps } from "@/types/header";
import Logo from "../Logo/Logo";
import url from "@/assets/images/js-logo4.png"
import Avatar from "../Avatar/Avatar";
import Menu from "../Menu/Menu";
import MENU_LIST from "@/constants/menu";
import "./header.css";

const Header: React.FC<HeaderProps> = ({ username, image }) => {
  return (
    <header className="header">
      <div className="header__logo">
        <Logo href="#Home" src={url} />
        <Menu menuList={MENU_LIST} />
      </div>
      <div className="header__user">
        <Avatar
          username={username}
          avatar={image}
        />
      </div>
    </header>
  );
}

export default Header;
