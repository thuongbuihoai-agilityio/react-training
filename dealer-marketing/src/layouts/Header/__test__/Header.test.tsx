import { render } from "@testing-library/react";
import Header from "../index";

describe("Header component", () => {
  test("Should render Header component", () => {
    const { getByTestId } = render(<Header />);
    const header = getByTestId("header");
    expect(header).toBeInTheDocument();
  });

  test("Matches snapshot", () => {
    const { container } = render(<Header />);
    expect(container).toMatchSnapshot();
  });
});
