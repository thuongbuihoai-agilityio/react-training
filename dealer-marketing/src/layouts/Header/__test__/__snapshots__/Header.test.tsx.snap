// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header component Matches snapshot 1`] = `
<div>
  <header
    data-testid="header"
  >
    <div>
      <div
        data-testid="menu"
      >
        <ul>
          <li>
            <a
              href="/expert-panel"
            >
              Our Experts
            </a>
          </li>
          <li>
            <a
              href="/about-us"
            >
              About
            </a>
          </li>
        </ul>
      </div>
    </div>
  </header>
</div>
`;
